Add support to automatically update the "Stable tag" and "Tested up to" lines in the readme.txt during deployment.

toolstack opened the pull request:
### Description of the Change
Add support to automatically update the "Stable tag" and "Tested up to" lines in the readme.txt during deployment.

Closes #112 

### How to test the Change
Trigger a new deployment.

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
Added - Support for updating readme.txt "Tested up to" and "Stable tag" automatically as part of the deployment.


### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @toolstack


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [] I have added tests to cover my change.
- [x] All new and existing tests pass.

toolstack committed:
Add support to update readme.txt

Both "Stable tag" and "Tested up to" are updated (if required) and committed back to the repo, along with the tag being updated to include the new version of the readme file.

toolstack committed:
Change commit message.

toolstack committed:
Update readme.

toolstack committed:
Cleanup the retrieval of the current wp version.

toolstack committed:
Add options to skip the new readme.txt updates.

jeffpaul commented on README.md:
```diff
@@ -23,6 +25,9 @@ This Action commits the contents of your Git tag to the WordPress.org plugin rep
 * `VERSION` - defaults to the tag name; do not recommend setting this except for testing purposes.
 * `ASSETS_DIR` - defaults to `.wordpress-org`, customizable for other locations of WordPress.org plugin repository-specific assets that belong in the top-level `assets` directory (the one on the same level as `trunk`).
 * `BUILD_DIR` - defaults to `false`. Set this flag to the directory where you build your plugins files into, then the action will copy and deploy files from that directory. Both absolute and relative paths are supported. The relative path if provided will be concatenated with the repository root directory. All files and folders in the build directory will be deployed, `.disignore` or `.gitattribu ...
+* `SKIP_WP_VERSION` - skips the retrieval of the current WP version from wordpress.org and does not update the readme.txt's "Tested up to" line.
```
I think my preference here would be the opposite here, that the default would NOT auto-bump the version without specific action by a maintainer.  As such, I'd say changing this to `BUMP_WP_TESTED_UP_TO` or something and that only when that is set to `true` (or otherwise present I guess) would it trigger the functionality in `deploy.sh` below.

jeffpaul replied:
Also, if someone forgot to properly bump versions then they could use https://github.com/10up/action-wordpress-plugin-asset-update to correct their mistake, right?

toolstack replied:
Sounds good, I've pushed an update to invert the logic.

toolstack committed:
Flip logic of WP Version update.

As per discussion, make the default to not update the WP Version during a release, but have a flag to enable it.

jeffpaul commented:
In re-reviewing this, I don't think that our primary use case (or even an optional case here) would be to update the plugin source in GH from values on WP.org.  I would expect that the values on GH should be updating WP.org (and not vice verse as I think this PR would do) and as such would recommend we close out this PR.  @dkotter thoughts?

dkotter commented:
> In re-reviewing this, I don't think that our primary use case (or even an optional case here) would be to update the plugin source in GH from values on WP.org. I would expect that the values on GH should be updating WP.org (and not vice verse as I think this PR would do) and as such would recommend we close out this PR. @dkotter thoughts?

Yes, I agree with that and also agree with your initial comment here: https://github.com/10up/action-wordpress-plugin-deploy/issues/112#issuecomment-1375985590. I think the most we should do here is have the deploy fail if the version number hasn't been bumped but I would not expect this action to automatically update that version and commit it back to GitHub.

toolstack commented:
> In re-reviewing this, I don't think that our primary use case (or even an optional case here) would be to update the plugin source in GH from values on WP.org. I would expect that the values on GH should be updating WP.org (and not vice verse as I think this PR would do) and as such would recommend we close out this PR. @dkotter thoughts?

This PR pulls the current WP version from WP.org (through the update api), and then updates GitHub before the push out to wp.org's plugin directory.  This ensure that both GH and WP's repo's are up to date.

toolstack commented:
> Yes, I agree with that and also agree with your initial comment here: [#112 (comment)](https://github.com/10up/action-wordpress-plugin-deploy/issues/112#issuecomment-1375985590). I think the most we should do here is have the deploy fail if the version number hasn't been bumped but I would not expect this action to automatically update that version and commit it back to GitHub.

I'm fine with not merging this if it's considered out of scope for the action.  I wouldn't want it to fail due to the non-updated value, that would seem heavy handed if the user wanted that to be the result (aka, perhaps the plugin doesn't work with newer versions of WP yet).

If the decision is to reject the change I'll fork the action (or maybe a separate action that just does the update) and release my own as I do think this is a useful feature to have.

jeffpaul commented:
@toolstack that sounds reasonable to me, thanks!

AlecRust commented:
> I'll fork the action (or maybe a separate action that just does the update) and release my own

Just FYI there are existing actions that can do this, [like mine](https://github.com/AlecRust/wp-tut-updater-action) 🙂 

My personal preference here is to manage all three things separately, "Stable tag" as part of the release process ([example](https://github.com/AlecRust/brevwoo/blob/8c043b876969e1bb74ebf7cfd31a7fb4b589edb9/.release-it.json#L7)) "Tested up to" as a PR that gets created, and the plugin deploy via this action. This action should just deploy imo.
